app/models/role.rb

```diff
@@ -37,32 +37,21 @@ def self.included(base)
       base.extend(ClassMethods)
       base.instance_eval do
         has_many :roles, :as => :authorizable
+        has_many :users, :through => :roles
 
-       scope :with_related_users_included, -> { includes(:roles => :users ) }
-       scope :of_interest_to, ->(user) {
-          {
-            :joins => joins_through_to_users,
-            :conditions => ['rj_u.id=?', user.id],
-            :group => "rj_r.authorizable_id"
-          }
+        scope :with_related_users_included, -> { includes(:roles => :users ) }
+        scope :of_interest_to, ->(user) {
+          joins(:users).where(users:{id:user}).uniq
```

Comment:
I think it should be 'user.id' like: joins(:users).where(users:{id:user.id}).uniq